library/std/src/ffi/c_str.rs

```diff
@@ -1266,34 +1266,36 @@ impl CStr {
     ///
     /// ```no_run
     /// # #![allow(unused_must_use)] #![allow(temporary_cstring_as_ptr)]
-    /// use std::ffi::CString;
+    /// use std::ffi::CStr;
     ///
-    /// let ptr = CString::new("Hello").expect("CString::new failed").as_ptr();
+    /// let ptr = {
+    ///     let hello = CStr::from_bytes_with_nul(b"Hello\0").expect("CStr::from_bytes_with_nul failed");
+    ///     hello.as_ptr()
+    /// };
     /// unsafe {
     ///     // `ptr` is dangling
     ///     *ptr;
     /// }
```

Comment:
Hmm, isn't this valid because `b"Hello\0"` is &'static? Or is it valid as long as the *CStr* is, not the inner value?

Reply:
You're right.  The lifetime of the `CStr` is that of the bytes passed to `from_bytes_with_nul`.  I'll revise.

Reply:
Good point!  Have used a byte array on the stack instead.